presto-native-execution/presto_cpp/main/JsonSignatureParser.cpp

```diff
@@ -57,25 +58,29 @@ velox::exec::FunctionSignaturePtr parseSignature(const folly::dynamic& input) {
       (outputType != nullptr) && (paramTypes != nullptr),
       "`outputType` and `paramTypes` are mandatory in a signature.");
 
+  auto* schema = input.get_ptr("schema");
+
   auto paramTypeSignatures = parseTypeSignatures(*paramTypes);
   std::vector<bool> constantArguments(paramTypeSignatures.size(), false);
 
-  return std::make_shared<velox::exec::FunctionSignature>(
-      std::unordered_map<std::string, velox::exec::SignatureVariable>{},
-      parseTypeSignature(*outputType),
-      std::move(paramTypeSignatures),
-      std::move(constantArguments),
-      /*variableArity=*/false);
+  return JsonSignatureParser::FunctionSignatureItem{
+      std::make_shared<velox::exec::FunctionSignature>(
+          std::unordered_map<std::string, velox::exec::SignatureVariable>{},
+          parseTypeSignature(*outputType),
+          std::move(paramTypeSignatures),
+          std::move(constantArguments),
+          /*variableArity=*/false),
+      (schema != nullptr) ? schema->asString() : ""};
```

Comment:
nit: Not sure how big schema is, but wont this create a copy of schema every time and possibly be expensive ? Would schema change for every functionsignatureitem ?

Reply:
Discussed offline. Schema here will likely be a small string